agent/dns_test.go

```diff
@@ -3832,6 +3832,45 @@ func TestDNS_Recurse_Truncation(t *testing.T) {
 	}
 }
 
+func TestDNS_Recursor_StrategyRandom(t *testing.T) {
+	configuredRecursors := []string{"1.1.1.1", "8.8.4.4", "8.8.8.8"}
+	recursorStrategy := agentdns.RecursorStrategy("random")
+
+	retries := func() *retry.Counter {
+		return &retry.Counter{Count: 5}
+	}
+
+	retry.RunWith(retries(), t, func(r *retry.R) {
```

Comment:
minor: no need for this function I think, we can specify the value directly

```suggestion
	retry.RunWith(&retry.Counter{Count: 5}, t, func(r *retry.R) {
```